src/base/searchengine.h

```diff
@@ -38,10 +40,119 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
```

Comment:
>How do you suggest to implement conversion to a string? Omit zeroed components too?

Is it the problem? There is common way: minimum items (default to, for example, 2) + omit trailing zeros.

Omitting trailing zeros is a common practice. But your implementation disallow it.